test/unit/io/sequence_file/sequence_file_integration_test.cpp

```diff
@@ -0,0 +1,165 @@
+// DAMAGE.
+//
+// ==========================================================================
+
+#include <sstream>
+
+#include <gtest/gtest.h>
+
+#include <seqan3/io/sequence_file/output.hpp>
+#include <seqan3/io/sequence_file/input.hpp>
+#include <seqan3/range/view/persist.hpp>
+#include <seqan3/range/view/take.hpp>
+#include <seqan3/test/tmp_filename.hpp>
+#include <seqan3/std/iterator>
+
+using namespace seqan3;
+using namespace seqan3::literal;
+
+TEST(rows, assign_sequence_files)
+{
+    std::string const input // differs from output above by formatting
+    {
+        ">TEST 1\n"
+        "ACGT\n"
+        "> Test2\n"
+        "AGGCTGN AGGCTGN AGGCTGN AGGCTGN AGGCTGN AGGCTGN AGGCTGN AGGCTGN AGGCTGN AGGCTGN AGGCTGN AGGCTGN AGGCTGN\n\n"
+        "> Test3\n"
+        "GGAGTATAATATATATATATATAT\n"
+    };
+
+    std::string const output_comp
+    {
+        "> TEST 1\n"
+        "ACGT\n"
+        "> Test2\n"
+        "AGGCTGNAGGCTGNAGGCTGNAGGCTGNAGGCTGNAGGCTGNAGGCTGNAGGCTGNAGGCTGNAGGCTGNAGGCTGNAGGCTGNAGGCTGN\n"
+        "> Test3\n"
+        "GGAGTATAATATATATATATATAT\n"
+    };
+
+    sequence_file_input fin{std::istringstream{input}, sequence_file_format_fasta{}};
+    sequence_file_output fout{std::ostringstream{}, sequence_file_format_fasta{}};
+    fout.options.fasta_letters_per_line = 0;
+
+    fout = fin;
+
+    fout.get_stream().flush();
+    EXPECT_EQ(fout.get_stream().str(), output_comp);
+}
+
+TEST(integration, assign_sequence_file_pipes)
+{
+    std::string const input
+    {
+        "> TEST1\n"
+        "ACGT\n"
+        "> Test2\n"
+        "AGGCTGNAGGCTGAGGCTGNAGGCTGNAGGCTGNAGGCTGNAGGCTGNAGGCTGNAGGCTGN\n"
+        "> Test3\n"
+        "GGAGTATAATATATATATATATAT\n"
+    };
+
+    // valid without assignment?
+    sequence_file_input{std::istringstream{input}, sequence_file_format_fasta{}} |
+        sequence_file_output{std::ostringstream{}, sequence_file_format_fasta{}};
+
+    // valid with assignment and check contents
+    auto fout = sequence_file_input{std::istringstream{input}, sequence_file_format_fasta{}} |
+                sequence_file_output{std::ostringstream{}, sequence_file_format_fasta{}};
+
+    fout.get_stream().flush();
+    EXPECT_EQ(fout.get_stream().str(), input);
+}
+
+TEST(integration, view)
+{
+    std::string const input
+    {
+        "> TEST1\n"
+        "ACGT\n"
+        "> Test2\n"
+        "AGGCTGNAGGCTGAGGCTGNAGGCTGNAGGCTGNAGGCTGNAGGCTGNAGGCTGNAGGCTGN\n"
+        "> Test3\n"
+        "GGAGTATAATATATATATATATAT\n"
+    };
+
+    std::string const output
+    {
+        "> TEST1\n"
+        "ACGT\n"
+        "> Test2\n"
+        "AGGCTGNAGGCTGAGGCTGNAGGCTGNAGGCTGNAGGCTGNAGGCTGNAGGCTGNAGGCTGN\n"
+    };
+
+    // valid without assignment?
+    sequence_file_input{std::istringstream{input}, sequence_file_format_fasta{}} | view::persist | view::take(2) |
+        sequence_file_output{std::ostringstream{}, sequence_file_format_fasta{}};
```

Comment:
... this doesn't? Is it because `view::take` is not rvalue aware?

Reply:
views don't take rvalues, but sequence_file_out is not a view :wink: